azerothcore/azerothcore-wotlk
# feat(DB): Set faction leaders and capital city guards to their vanilla stats

This was a suggestion for chromiecraft as it will encourage world pvp.
This diff will have 2 parts: Changing the faction leaders, and changing all the guards in the capital cities to be lvl 60.
Ideally, this diff would be behind some configuration flag so that any server could choose whether or not to use the vanilla or wotlk values.
I used wowhead to get their classic HP values, e.g. https://classic.wowhead.com/npc=4949/thrall . For faction leaders added in TBC I just scaled them equally.
I tried testing the level, as it just shows ?? Boss and I didn't find a GM command to show the level, I made my character level 60, learned all weapon skills and hit an unedited faction leader (lv 83) and a modified one, but I didn't even miss on the 83 one. I thought a level 60 was supposed to miss basically every hit on a 83 boss?
I did not verify yet if their abilities are dealing lv 80 damage or lv 60.

To get the IDs I did the following: Searched creature_template by name, verified by using .go creature <ID> command that it's the right NPCs, adjusted the values, then spawned the faction leaders to check their values.



## Changes Proposed:
-  Sets all faction leaders to lvl 63 and their vanilla HP values in order to give another world PvP objective on chromie
-  In next commit: Set all guards in capital cities to level 63

## Issues Addressed:

-

## SOURCE:


## Tests Performed:

- DB change only. Tested by spawning them in-game (to avoid teleporting 10 times). Verified that the IDs are correct by teleporting to each one beforehand.
![grafik](https://user-images.githubusercontent.com/5873110/126094750-8d6eb54b-8a4b-467d-ac9b-c3787c0592b2.png)

Sample screenshot of test, I have one for each faction leader (ignore the changed name of Varian, this is my first PR and that was my first change :) )
![WoWScrnShot_071921_031034](https://user-images.githubusercontent.com/5873110/126094944-76f8fa46-0ab0-4c08-9a76-49a29aeafeb8.jpg)


## How to Test the Changes:


1. Apply SQL changes
2. Restart world server
3. TP to any of the creatures from the table above, verify that they have 60 stats.

## Known Issues and TODO List:


- [TODO] Verify that their combat strength is appropriate for 60s
- [TODO] Change guards in capital cities to be 62 as well (currently: 82)


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
